src/panel.ts

```diff
@@ -114,26 +116,56 @@ export class HistoryDebuggerPanel {
       // TODO: error handling
       switch (e.type) {
         case "startFromId": {
-          console.log()
-          await vscode.window.showInformationMessage("Starting debug session")
+          // TODO: Implement history pagination
+          const conn = await Connection.connect(/* { address: 'temporal.prod.company.com' } */)
+          const { history } = await conn.workflowService.getWorkflowExecutionHistory({
+            namespace: "default",
+            execution: {
+              workflowId: e.workflowId,
+              runId: e.runId,
+            },
+          })
+          if (!history) {
+            await vscode.window.showErrorMessage("Empty history")
+            break
+          }
+          const bytes = new Uint8Array(temporal.api.history.v1.History.encodeDelimited(history).finish())
+          this.currentHistoryBuffer = bytes
+          await this.handleStartProject(bytes)
           break
         }
         case "startFromHistory": {
           const buffer = Buffer.from(e.buffer)
           // TODO: support binary history too
           const history = historyFromJSON(JSON.parse(buffer.toString()))
           const bytes = new Uint8Array(temporal.api.history.v1.History.encodeDelimited(history).finish())
-          this.currentHistoryBuffer = buffer
-          await webview.postMessage({ type: "historyProcessed", history: bytes })
-          // eslint-disable-next-line  @typescript-eslint/naming-convention
-          const _config = { env: { TEMPORAL_DEBUGGER_PLUGIN_URL: this.httpServerUrl } }
-          await vscode.window.showInformationMessage("Starting debug session")
+          this.currentHistoryBuffer = bytes
+          await this.handleStartProject(bytes)
           break
         }
       }
     })
   }
 
+  private async handleStartProject(bytes: Uint8Array): Promise<void> {
```

Comment:
We should have moved setting the currentHistoryBuffer to this function too but we can fix later, I don't want to block this PR anymore.